Tools/LambdaTestTool-v2/src/Amazon.Lambda.TestTool/Utilities/RouteTemplateUtility.cs

```diff
@@ -0,0 +1,66 @@
+﻿namespace Amazon.Lambda.TestTool.Utilities;
+
+using Microsoft.AspNetCore.Routing.Template;
+
+/// <summary>
+/// Provides utility methods for working with route templates and extracting path parameters.
+/// </summary>
+public static class RouteTemplateUtility
+{
+    /// <summary>
+    /// Extracts path parameters from an actual path based on a route template.
+    /// </summary>
+    /// <param name="routeTemplate">The route template to match against.</param>
+    /// <param name="actualPath">The actual path to extract parameters from.</param>
+    /// <returns>A dictionary of extracted path parameters and their values.</returns>
+    /// <example>
+    /// Using this method:
+    /// <code>
+    /// var routeTemplate = "/users/{id}/orders/{orderId}";
+    /// var actualPath = "/users/123/orders/456";
+    /// var parameters = RouteTemplateUtility.ExtractPathParameters(routeTemplate, actualPath);
+    /// // parameters will contain: { {"id", "123"}, {"orderId", "456"} }
+    /// </code>
+    /// </example>
+    public static Dictionary<string, string> ExtractPathParameters(string routeTemplate, string actualPath)
```

Comment:
This does not take into account the greedy variable {proxy+}

Reply:
this should be fixed now